### PR TITLE
PCHR-3494: Fix CRM modals position relatively to CiviHR modals

### PR DESCRIPTION
**Pull request by igorpavlov-zz**
## Overview

This PR makes CRM modals appear on top of CiviHR modals.

## Technical Details

CiviHR modals have z-indexes starting from 1040 and are incremented by 10 within each stacked modal.

Z-index 2000 was selected for the CRM modals, that will allow 96 CiviHR modals to be stacked before they appear on top of the CRM modals, which is extremely unlikely.

A fixed z-index for CRM modal is not a problem since they all appear at the same DOM level (just before closing `</body>`, so they will stack together nicely even with the same `z-index`.

```css
/* stylelint-disable max-nesting-depth, selector-max-compound-selectors, selector-max-id */

.crm-container.ui-dialog {
  // ...
  z-index: 2001;

// ...

.ui-widget-overlay.ui-front {
  z-index: 2000;
}
```

_Here how modals stack:_

ℹ️Please note, it is considered that the CiviHR modal will not be likely called from CRM modal, which makes the situation when we need CiviHR modal on top of CRM modal very unlikely.

### Outside-namespace

`outside-namespace` structure added. Now, in order to not wrap selectors inside `#bootstrap-theme` you have 2 options:

#### `.____outside-namespace` selector

```css
// Use wrapping selector
// handy for particular rules inside a file with other rules,
// which need theme wrapping
.___outside-namespace {
   a { color: red } // will make all links red
}
a { color: yellow } // will make links inside `#bootstrap-theme` yellow
```

#### Specific `outside-namespace` folders
Locations: 
- scss/bootstrap/overrides/crm/outside-namespace
- scss/bootstrap/components/outside-namespace

```css
// Put a file into `outside-namespace` folder
// handy for batch rules that are all outside the theme
a { color: red } // will make all links red
```

### Minor changes

- A typo in README file was fixed.

------

✅Manual Tests - passed
✅CSS distributive files - included
⚠️Backstop tests - existing passed, but new suite cannot be created due to https://github.com/civicrm/civihr/pull/2536 since buttons have been changed

### CODE REVIEW

File: scss/civicrm/common/_ui-dialog.scss [hunk 1]
```diff
@@ -1,15 +1,18 @@
+/* stylelint-disable max-nesting-depth, selector-max-compound-selectors, selector-max-id */
```
Comment (AkA84) [inline scss/civicrm/common/_ui-dialog.scss]: where did you get this? In CODING.md it's
```scss
/* stylelint-disable max-nesting-depth, selector-max-compound-selectors, selector-no-qualifying-type, selector-max-id */
```

File: scss/civicrm/common/_ui-dialog.scss [hunk 2]
```diff
@@ -43,8 +46,12 @@
...
 
   .ui-dialog-content#bootstrap-theme,
-  .ui-dialog-content[id*="crm-ajax-dialog"] {
+  .ui-dialog-content[id*='crm-ajax-dialog'] {
     background: $crm-white;
     padding: 10px 20px;
   }
 }
+
+.ui-widget-overlay.ui-front {
```
Comment (AkA84) [inline scss/civicrm/common/_ui-dialog.scss]: This PR is about making sure that the old UI modal appear on top of *bootstrap* modals, meaning that all the CSS changes should be in the bootstrap-side of the theme (bootstrap/ folder), not in the legacy side (civicrm/ folder).

There is a [bootstrap/overrides/crm](https://github.com/compucorp/org.civicrm.shoreditch/tree/staging/scss/bootstrap/overrides/crm) folder explicitly for overriding crm styles when inside the bootstrap namespace

**Review by AkA84 — CHANGES_REQUESTED** [review]

File: gulpfile.js [hunk 3]
```diff
@@ -21,9 +22,10 @@ gulp.task('sass:bootstrap', ['sass:sync'], function () {
     .pipe(stripCssComments({ preserve: false }))
     .pipe(postcss([postcssPrefix({
       prefix: bootstrapNamespace + ' ',
-      exclude: [/^html/, /^body/, /\.ta-hidden-input/, /\.mobile/]
+      exclude: [/^html/, /^body/, /\.ta-hidden-input/, /\.mobile/, outsideNamespaceRegExp]
```
Comment (AkA84) [inline gulpfile.js]: Now that we have this special namespace class, i think we can get rid of `.ta-hidden-input` and `.mobile` in this list, if we wrapped it in the namespace

**Review by AkA84 — CHANGES_REQUESTED** [review]
@igorpavlov please update the PR description with infor about the new namespace

File: gulpfile.js (hunk 3 above)
Comment (igorpavlov-zz) [reply gulpfile.js]: @AkA84 `.ta-hidden-input` is inside vendor file, not sure I can do anything with it.
`./scss/bootstrap/vendor/text-angular/_text-angular.scss`

File: scss/bootstrap/components/outside-namespace/_date-picker.scss [hunk 4]
```diff
@@ -1,10 +1,12 @@
-@at-root .mobile {
-  input[type="date"][uib-datepicker-popup] {
+.mobile {
+  [type='date'][uib-datepicker-popup] {
     line-height: normal;
 
     &::-webkit-inner-spin-button,
     &::-webkit-clear-button {
+      /* stylelint-disable */
       -webkit-appearance: none;
```
Comment (AkA84) [inline scss/bootstrap/components/outside-namespace/_date-picker.scss]:
```scss
-webkit-appearance: none; /* stylelint-disable-line {whatever-rule} */

File: scss/bootstrap/overrides/crm/outside-namespace/_ui-select.scss [hunk 5]
```diff
@@ -0,0 +1,7 @@
+.crm-container.ui-dialog {
```
Comment (AkA84) [inline scss/bootstrap/overrides/crm/outside-namespace/_ui-select.scss:7]: Leave a comment here explaining what this partial is doing

**Review by AkA84 — CHANGES_REQUESTED** [review]

**Review by AkA84 — APPROVED** [review]

**Pull request merged**